### PR TITLE
Bulk edit locations when default conferencing app is set 

### LINKED ISSUE #7034: [CAL-1039] Bulk apply default conferencing app to already created events
As a user with many existing event types, I want to be able to bulk apply my default to those existing events.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1039](https://linear.app/calcom/issue/CAL-1039/bulk-apply-default-conferencing-app-to-already-created-events)</sub>

### PR DESCRIPTION
**Pull request by sean-brydon**
Bulk edit locations for all existing event types to allow the user to update all current locations to a new default. 

Loom: https://www.loom.com/share/9255db32457c4f659c2dbc7f6eefb9d5

Fixes: https://github.com/calcom/cal.com/issues/7034

Although deems low priority this was confusing a lot of users (see threads for more convo on this) as this is what they expected it did by default. However not all users would want this (so came up with the idea to easily select what you want to update) 

TODO: Reuse this in appstore and not just settings 

CC: @Jaibles the design and flow hasnt been reviewed by you yet - please leave feedback in [figma](https://www.figma.com/file/UajFu4M1APhn2AywAEJkJr/Cal---New-Features?node-id=3266%3A172033&t=LXgqSsKclPFeOBZH-4) or loom :)

### CODE REVIEW

File: packages/trpc/server/routers/viewer/eventTypes.ts [hunk 1]
```diff
@@ -829,6 +830,70 @@ export const eventTypesRouter = router({
...
+    });
+
+    const eventTypesWithLogo = eventTypes.map((eventType) => {
+      const locationParsed = eventTypeLocationsSchema.parse(eventType.locations);
+      const app = getAppFromLocationValue(locationParsed[0].type);
+      return {
+        ...eventType,
+        logo: app?.logo,
+      };
+    });
```
Comment (sean-brydon) [inline packages/trpc/server/routers/viewer/eventTypes.ts:853]: Do this server side much easier than requiring credentials to do this

File: packages/trpc/server/routers/viewer/eventTypes.ts [hunk 2]
```diff
@@ -829,6 +830,70 @@ export const eventTypesRouter = router({
...
+
+      return await ctx.prisma.eventType.updateMany({
+        where: {
+          id: {
+            in: eventTypeIds,
+          },
+          userId: ctx.user.id,
+        },
+        data: {
+          locations: [{ type: appType, link: defaultApp.appLink }] as LocationObject[],
```
Comment (sean-brydon) [inline packages/trpc/server/routers/viewer/eventTypes.ts:893]: Override all locations with default

**Review by emrysal — COMMENTED** [review]
From a tech perspective it's good, but it's also a UX thing so deferring to @Jaibles

**Review by emrysal — APPROVED** [review]
Approved, we can always amend the design @Jaibles 🙏

**Pull request merged**